contracts/FIFOMarket.sol

```diff
@@ -536,6 +511,28 @@ contract FIFOMarket is
     return true; // returns true if the value was added to the set, that is, if it was not already present
   }
 
+  // todo consider making this a generalized `withdrawRemoval`?
+  function release(uint256 removalId, uint256 amount) external returns (bool) {
+    // todo whenNotPaused
+    // todo require sender is Removal address
+    address supplierAddress = removalId.supplierAddress();
+    uint256 removalBalance = _removal.balanceOf(address(this), removalId);
+    if (amount == removalBalance) {
```

Comment:
I see that the way this is called from within `Removal.release()` is that the full amount being released is passed to this function as well.   It seems to me that there's a difference though between 1) creating a record of the full amount of this removal that has been released (so that's what's going in the `_tokenIdToReleased` map) and 2) performing the necessary actions to remedy things, such as burning any possible unsold balance of that removal, and then, if there wasn't enough unsold balance, making the replacements to any certificates that had consumed the released removal (not in scope for this PR)

All that to say that I think this if check should actually be asking if `(amount >= removalBalance)` instead of `==`, which allows us to still pass the full released amount, recognizing that it might actually be overshooting the unsold removal balance, in which case we still want to pull the removal id out of any supply pools we have.

Reply:
@amiecorso is this still an issue in the latest commit?

Reply:
discussed on a call and you're addressing in #305